utils/backoff.go

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) 2016-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package utils
+
+import (
+	"time"
+)
+
+const (
+	backoffBase uint64 = 128
+	maxAttempts        = 4
+)
+
+// BackoffOperation is executed by Retry.
+// The BackoffOperation will be retried the provided
+// number attempts if returning an error.
+type BackoffOperation func() error
+
+// ProgressiveRetry executes a BackoffOperation and retries the operation 3 times upon error.
+func ProgressiveRetry(bo BackoffOperation) error {
```

Comment:
nit: 1/5 s/bo/tryf/ (or whatever, to match the ^^ type name?)

Reply:
Might be a dumb question, but what do you mean with `1/5` and `0/5`?

Reply:
Heh, secret decoder ring required: https://docs.mattermost.com/process/training.html#id8.